src/Traits/ModifyExtendTrait.php

```diff
@@ -0,0 +1,56 @@
+<?php
+/**
+ * Copyright (c) Cake Software Foundation, Inc. (http://cakefoundation.org)
+ *
+ * Licensed under The MIT License
+ * Redistributions of files must retain the above copyright notice.
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (http://cakefoundation.org)
+ * @link          http://cakephp.org CakePHP(tm) Project
+ * @license       http://www.opensource.org/licenses/mit-license.php MIT License
+ */
+namespace Cake\Chronos\Traits;
+
+/**
+ * A trait for modify method extending.
+ *
+ * Used in date objects, both mutable and immutable.
+ * Shoudn't used with FrozenTimeTrait because they both overwrites modify method.
+ */
+trait ModifyExtendTrait
+{
+
+    /**
+     * In addition to original modify method:
+     *     - handling modification with DST change (see comments in code)
+     *
+     * @param string $modify
+     * @return static
+     */
+    public function modify($modify)
+    {
+        // If we changing time in some timezone with daylight correction, there can be the case,
+        // that this correction influence to time in nonwanted way.
+        // So we have to make this modification in UTC timezone.
+        
+        if (!preg_match('/(sec|second|min|minute|hour)s?/i', $modify)) {
+            return parent::modify($modify);
+        }
+
+        $timezone = $this->getTimezone();
+        $timezoneName = $timezone->getName();
+
+        // Here is tricky code for immutable DateTime
+        // I use parent modify method only if current object timezone is UTC
+        // If it is not - i making another object in UTC wich will call to parent modify
+        if ($timezoneName !== 'UTC') {
+            $date = $this->setTimezone('UTC');
```

Comment:
Yeah, i know. I think that my way is more clear. But it is quite "matter of opinion" question, so I will chain it in that case because there only three operations.
Thank you.